From bad7564349b211c3bef2d497a2ae89106dee02c6 Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Sat, 22 Oct 2022 21:11:35 +0800
Subject: [PATCH] vulkan:fix buf map failure in 32bits app

1) fix buf map failure in 32bits app
2) and when has_tiling_uapi is false
   return I915_TILING_NONE which means ISL_TILING_LINEAR
   On mesa, it returns -1, will cause vulkan app always crash
   has_tiling_uapi is always false, because DRM_IOCTL_I915_GEM_SET_TILING
   is not supported in has_get_tiling() function

commit 1de5d2ac010502913777086849e4b4dabbf89d57
Signed-off-by: Chen, Yu <yu.y.chen@intel.com>
Change-Id: I94fe62080909841aff2ec2d57f942453989e11e1
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 src/intel/vulkan/anv_gem.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/intel/vulkan/anv_gem.c b/src/intel/vulkan/anv_gem.c
index 80b45e344c5..5f00dc9fd8d 100644
--- a/src/intel/vulkan/anv_gem.c
+++ b/src/intel/vulkan/anv_gem.c
@@ -113,8 +113,12 @@ anv_gem_mmap_offset(struct anv_device *device, uint32_t gem_handle,
       return MAP_FAILED;
 
    /* And map it */
-   void *map = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED,
-                    device->fd, gem_mmap.offset);
+   #ifdef __x86_64__
+   void *map = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED, device->fd, gem_mmap.offset);
+   #else
+   void *map = mmap64(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED, device->fd, gem_mmap.offset);
+   #endif
+
    return map;
 }
 
@@ -232,7 +236,7 @@ int
 anv_gem_get_tiling(struct anv_device *device, uint32_t gem_handle)
 {
    if (!device->info->has_tiling_uapi)
-      return -1;
+      return I915_TILING_NONE;
 
    struct drm_i915_gem_get_tiling get_tiling = {
       .handle = gem_handle,
-- 
2.39.2

